tensorflow_model_optimization/python/core/sparsity_tf2/pruner_test.py

```diff
@@ -0,0 +1,493 @@
+  end_epoch: total number of epochs to train for where pruning is eligible
+
+  Returns:
+    percent to prune to after each cyle/round
+  """
+  # no pruning until weights are saved TODO: off by one??
+  frequency = tf.math.floordiv(end_epoch - save_round + 1, n_rounds) # range(0, end, freq)
+  prune_pcent_per_round = tf.math.pow(target_sparsity, tf.math.divide(1, n_rounds))
+  return tf.cast(frequency, tf.int64), prune_pcent_per_round
+
+def make_pruning_schedule(target_sparsity, begin, end, freq):
+  return pruning_schedule.ConstantSparsity(target_sparsity, begin, end, freq)
+
+def sample_noise(x, mu=0, sigma=1.):
+  dist = tf.random.normal(loc=mu,  sccale=sigma)
+  return dist.prob(x)
+
+class PruningTest(test.TestCase, parameterized.TestCase):
+
+  def setUp(self):
+    super(PruningTest, self).setUp()
+    self.block_size = (1, 1)
+    self.block_pooling_type = "AVG"
+    self.target_sparsity = 0.5
+    self.constant_sparsity = pruning_schedule.ConstantSparsity(self.target_sparsity, 0, 100, 1)
+    self.save_init = tf.Variable(0)
+    self.save_itr_10 = 10
+    # dummy optimizer
+    self.dummy_optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    self.grad = lambda shape: tf.ones_like(shape)
+
+  # Variable initialization outside of setUp() is needed for compatibility with
+  # run_all_keras_modes.
+  #
+  # setUp() lies outside of the "eager scope" that wraps the test cases
+  # themselves, resulting in initializing graph tensors instead of eager
+  # tensors when testing eager execution.
+  def initialize(self):
+    self.global_step = tf.Variable(
+        tf.zeros([], dtype=dtypes.int32),
+        dtype=dtypes.int32,
+        # dtype=tf.float64,
+        name="global_step")
+
+    def training_step_fn():
+      return self.global_step
+    self.training_step_fn = training_step_fn
+
+    compat.initialize_variables(self)
+
+  # def testUpdateSingleMask(self):
+  #   weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+  #   weight_dtype = weight.dtype.base_dtype
+  #   mask = tf.Variable(
+  #       tf.ones(weight.get_shape(), dtype=weight_dtype),
+  #       name="mask",
+  #       dtype=weight_dtype)
+  #   threshold = tf.Variable(
+  #       tf.zeros([], dtype=weight_dtype), name="threshold", dtype=weight_dtype)
+  #   self.initialize()
+  #   pruning_vars = [(weight, mask, threshold)]
+  #   next_step = self.training_step_fn() + 1
+
+  #   p = pruner.LTHPruner(
+  #       pruning_schedule=self.constant_sparsity,
+  #       save_schedule=self.save_init,
+  #       block_size=self.block_size,
+  #       block_pooling_type=self.block_pooling_type)
+
+  #   mask_before_pruning = K.get_value(mask)
+  #   self.assertAllEqual(np.count_nonzero(mask_before_pruning), 100)
+
+  #   if tf.executing_eagerly():
+  #     p.update_masks(pruning_vars, next_step)
+  #   else:
+  #     K.get_session().run(p.update_masks(pruning_vars, next_step))
+
+  #   mask_after_pruning = K.get_value(mask)
```

Comment:
Do `mask.numpy()` if you need the numpy value

Reply:
This applies to all such places in the tests.
Wherever K.get_value is used, get rid of it.
If you *have* to have the numpy value do `x.numpy()`. Otherwise don't do anything.